### ISSUE TITLE
CONTENT_LENGTH_MISMATCH for js file lazy loaded (Vite + React)

### ISSUE THREAD

**Issue by byronwall** [wallabyjs/console-ninja#48]
Console Ninja appears to break the importing/loading process of the Monaco editor.  It causes a CONTENT_LENGTH_MISTMATCH error to occur when the `editor.main.js` file is loaded.

I confirmed that disabling Console Ninja causes this to work again.  And it worked before.

I am running a Vite app.  I have the necessary Monaco files in my `public` folder.  I am using the `@monaco-editor/react` package to get the editor to appear.  I basically expect these js files to be available so the Monaco editor can load them or whatever it does.

I assume Console Ninja is modifying these files in transit somehow?  I further assume that the Vite dev server is specifying a content length that Console Ninja affects?  If so, it'd be great if there is some way to exclude files from the Console Ninja treatment.  In this case, I only want to activate on files in my `src/` tree instead of `public/`.  

For reference, both the `loader.js` and `editor.main.js` files contain `console.XXX` commands which presumably are being rewritten?

I am mostly guessing about the mechanism here.  Please let me know if I can provide more useful information.

![image](https://user-images.githubusercontent.com/495404/207412455-ea84e570-a2ef-427a-a733-c98e8b70bf46.png)

**Comment by ArtemGovorov** [comment]
> I assume Console Ninja is modifying these files in transit somehow?

Yes, your assumption [is correct](https://github.com/wallabyjs/console-ninja#how-does-it-work). 

> I further assume that the Vite dev server is specifying a content length that Console Ninja affects?

Console Ninja takes care of providing correct content length.

> If so, it'd be great if there is some way to exclude files from the Console Ninja treatment. In this case, I only want to activate on files in my src/ tree instead of public/.

Console Ninja is trying to show the same output that your Dev Tools console is showing. While I think it is a good idea to add the setting to exclude files from being touched by Console Ninja, it may make it work differently from Dev Tools. In your case however, it should just work without having to exclude any files (once the issue that you're observing is fixed).

We have tried to reproduce the issue by setting up a repo with the same structure/files, but it seems to be working for us. Having said that, we have fixed few related issues yesterday that may affect your scenario.

Can you please try the following:
- check your Console Ninja extension version, and if it's less than v0.0.35, please update to the latest version, restart vite and refresh the page to see if the issue is still there.
- clone https://github.com/wallabyjs/console-ninja-48 repo, open the `vs2/editor/editor.main.js` file in editor (this is to make 100% sure that Console Ninja will be instrumenting it), run `npm i`, run `npm run dev`, open the app in browser, check if the app is working or has the same content length error.
- if the sample is not working for you the same way as your real app, can you please share your OS and node version?
- if the sample is working for you as expected, can you please try adding things that we're missing to reproduce to it?

**Comment by byronwall** [comment]
@ArtemGovorov , thanks for the detailed response and follow up.  I'll test things out in the morning (my morning at least!) when I'm back at work.

**Comment by byronwall** [comment]
> Can you please try the following:
> check your Console Ninja extension version, and if it's less than v0.0.35, please update to the latest version, restart vite and refresh the page to see if the issue is still there.

I am using version `v0.0.43`. I had an upgrade waiting for me this morning, so I restarted VS Code with the new version.  I did not check the previous version being used before the upgrade.  It would have been whatever was released yesterday (Dec 13 2022) since I reset VS Code a couple times for unrelated things.

After that upgrade, it appears that things are working.  In addition to the new Console Ninja version, I did the following a couple of times:
- Stopped and started Console Ninja
- Refreshed the page
- Restarted the Vite dev server

Before restarting the Vite server, I was getting a number of errors related to web sockets and other stuff.  I had the same Vite instance running through the extension update.

After restarting Vite, the errors went away and the Monaco editor started working as expected.

I think at this point, I am in the camp of "it seems to work".  I'll let you know if something changes with a new issue.

Thanks again for the quick reply.
